src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/ObservationMapper.java

```diff
@@ -166,24 +171,34 @@ private void mapStatus(final LabResult labResult, final Observation observation)
             observation.setStatus(status);
         }
 
-        private void mapValueQuantity(final LabResult labResult, final Observation observation) {
+        private void mapInvestigationResult(final LabResult labResult, final Observation observation) {
             // Observation.value.valueQuantity.*
             labResult.getInvestigationResult().ifPresent(investigationResult -> {
-                final var quantity = new Quantity();
-
-                // Observation.value.valueQuantity.value = SG18.RSL.C830(1).6314
-                quantity.setValue(investigationResult.getMeasurementValue());
-
-                // Observation.value.valueQuantity.unit = SG18.RSL.C848.6410
-                quantity.setUnit(investigationResult.getMeasurementUnit());
-
-                // Observation.value.valueQuantity.comparator = SG18.RSL.C830(1).6321
-                investigationResult.getMeasurementValueComparator()
-                    .map(MeasurementValueComparator::getDescription)
-                    .map(QuantityComparator::fromCode)
-                    .ifPresent(quantity::setComparator);
-
-                observation.setValue(quantity);
+                if (investigationResult.getResultType() == LaboratoryInvestigationResultType.NUMERICAL_VALUE) {
```

Comment:
Have you added unit tests covering the two new `if statements`?

Reply:
Done 👍